x-pack/plugin/otel-data/src/main/resources/index-templates/logs-otel@template.yaml

```diff
@@ -0,0 +1,21 @@
+---
+version: ${xpack.oteldata.template.version}
+index_patterns: ["logs-*.otel-*"]
+priority: 120
+data_stream: {}
+allow_auto_create: true
+_meta:
+  description: default logs template installed by x-pack
+  managed: true
+composed_of:
+  - logs@mappings
```

Comment:
This would allow us to share very basic mappings and not duplicate those. 

In this specific one, the only thing that we pull in is to set [`data_stream.type` to ` "value": "logs"`](https://github.com/elastic/elasticsearch/blob/86c41c6670d72986ac07116bf911e8c274b10a6b/x-pack/plugin/core/template-resources/src/main/resources/logs%40mappings.json#L11) - the remaining things from `logs@mappings` are already covered in `otel@mappings`. I'd say setting `data_stream.type` to ` "value": "logs"` is something which makes sense to share across log related templates.

I don't have much history on how often these mappings from core are changed - or if there would be any risk here, but I think this would be a good way to share these very basic mappings and also use it for OTel.

We discussed that potentially we could also share a few things for traces between otel and `apm-data` - that'd be also done via a mapping in `core`. But that's outside this PR.

Reply:
That's an outcome from this discussion: https://github.com/elastic/elasticsearch/pull/111091#discussion_r1710769086.